pkg/gcv/result.go

```diff
@@ -34,41 +34,32 @@ const (
 
 // Result is the result of reviewing an individual resource
 type Result struct {
-	// The name of the resource as given by CAI
+	// The name of the resource as given to Config Validator
 	Name string
-	// CAIResource is the resource as given by CAI
-	CAIResource map[string]interface{}
+	// InputResource is the resource as given to Config Validator. This may be a
+	// CAI Asset or a Terraform Resource Change.
+	InputResource map[string]interface{}
 	// ReviewResource is the resource sent to Constraint Framework for review.
-	// For GCP types this is the unmodified resource from CAI, for K8S types, this is the unwrapped
-	// resource.
+	// This may be a CAI Asset, K8S resource, or Terraform Resource Change.
 	ReviewResource map[string]interface{}
 	// ConstraintViolations are the constraints that were not satisfied during review.
 	ConstraintViolations []ConstraintViolation
 }
 
 // NewResult creates a Result from the provided CF Response.
 func NewResult(
-	target string,
-	caiResource map[string]interface{},
+	target, name string,
+	inputResource map[string]interface{},
 	reviewResource map[string]interface{},
 	responses *cftypes.Responses) (*Result, error) {
 	cfResponse, found := responses.ByTarget[target]
 	if !found {
 		return nil, errors.Errorf("No response for target %s", target)
 	}
 
-	resNameIface, found := caiResource["name"]
-	if !found {
-		return nil, errors.Errorf("result missing name field")
-	}
-	name, ok := resNameIface.(string)
-	if !ok {
-		return nil, errors.Errorf("failed to convert resource name to string %v", resNameIface)
-	}
-
 	result := &Result{
 		Name:                 name,
-		CAIResource:          caiResource,
+		InputResource:        inputResource,
```

Comment:
won't this renaming of this struct parameters have implications on downstream consumers? I worry about making such an adjustment

Reply:
Yes, it could have implications on downstream consumers; however, I believe it's a necessary change since we're fundamentally changing the contract that config-validator has, by adding a type of input resource that isn't a CAI asset.

I think the primary entrypoints for consumers of this library are going to be `ReviewAsset` and `ReviewTFResourceChange`, which return a list of Violation objects - for those entrypoints, this change would not be noticeable.

Reply:
Yeah, I took a glance through https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/cli/scorecard/ and it doesn't look like we would break anything with this.